src/core/email-client.js

```diff
@@ -53,41 +68,154 @@ function getFreshImap (cb) {
 }
 
 class EmailClient {
-    constructor () {
+    //constructor takes the full name of user as String
+    constructor (name) {
         var self = this;
+        self._name = name; 
+        self._email = "";
 
         //  Get Imap obj and check emails
-        getFreshImap(function (err, imap) {
+        getFreshImap(function (err, imap, email, smtpConfig) {
             if (err) {
                 console.log(err);
             } else {
-                self.imap = imap;
+                self._imap = imap;
+                self._email = email;
+                self._transporter = NodeMailer.createTransport(smtpConfig);
                 self._checkEmails();
+                
+                
             }
         });
     }
 
+    //method to change name of user
+    _changeName(name){
+        var self = this. 
+        self._name = name;
+    }
+
+    //attachments as array of Path
+    _sendEmail(toName, toEmail, subject, text, attachments){
+        var self = this;
+
+        if (self._email == ""){
+            fs.readFile('./email.json', 'utf8', function (err, data) {
+                if (err) {
+                    cconsole.log("Error: ", err);
```

Comment:
You'll be able to catch syntax errors like this if you run lint on your code with grunt (check readme)